views/components/ShareCharts.jsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import Select from '@material-ui/core/Select'
+import Dialog from '@material-ui/core/Dialog'
+import DialogActions from '@material-ui/core/DialogActions'
+import DialogContent from '@material-ui/core/DialogContent'
+import DialogTitle from '@material-ui/core/DialogTitle'
+import Chip from '@material-ui/core/Chip'
+import MenuItem from '@material-ui/core/MenuItem'
+import Typography from '@material-ui/core/Typography'
+import Button from '@material-ui/core/Button'
+import Input from '@material-ui/core/Input'
+
+const menuProps = {
+  PaperProps: {
+    style: {
+      marginTop: '50px',
+    },
+  },
+}
+
+const ShareChart = ({
+  open,
+  chart,
+  userList,
+  handleChange,
+  handleDelete,
+  shareWithUsers,
+  closeDialog,
+}) => {
+  return (
+    <Dialog open={open} onClose={closeDialog} fullScreen={true}>
+      <DialogTitle
+        id='alert-dialog-title'
+        disableTypography={true}
+        style={{
+          backgroundColor: 'rgba(0,0,0,0.7)',
+        }}
+      >
+        <Typography
+          variant='title'
+          style={{
+            color: '#ffffff',
+          }}
+        >
+          Share your chart
+        </Typography>
+      </DialogTitle>
+      <DialogContent
+        style={{
+          padding: '24px',
+          overflowY: 'visible',
+        }}
+      >
+        <Select
+          multiple
+          value={chart.sharedWith || []}
+          onChange={handleChange}
+          input={<Input id='select-multiple' />}
+          MenuProps={menuProps}
+          fullWidth
+          renderValue={(selected) => (
+            <>
+              {selected.map((value) => (
+                <Chip
+                  key={value}
+                  label={value}
+                  onDelete={() => handleDelete(value)}
+                  value={value}
+                  onMouseDown={(event) => {
+                    event.stopPropagation()
+                  }}
+                />
+              ))}
+            </>
+          )}
+        >
+          {userList.map(({ value, label }) => (
+            <MenuItem key={value} value={value}>
```

Comment:
shouldn't we be rendering `option` elements inside a `select` and not `li` elements?

Reply:
The documentation for material uses menu items.
https://v3.mui.com/demos/selects/